sequencer/finalizer.go

```diff
@@ -427,3 +431,27 @@ func (f *finalizer) WaitUntilResults(ctx context.Context, pid *types.ProcessID)
 		}
 	}
 }
+
+// cleanStaleVotesByProcessID removes all votes and related data for a given
+// process ID. It cleans up pending ballots, verified ballots, aggregated
+// ballots, and pending state transition batches associated with the process.
+func (f *finalizer) cleanStaleVotesByProcessID(pid []byte) error {
```

Comment:
I think this cleaning should not only happen when finalize() is completed, but when a voting process goes to the RESULTS status. Because it could be that other Sequencer on the network finalize it, and we still need to clean.